File: 1_Review/julia/5_importing_custom_module.jl
# 5) Importing a custom module
# Run: julia 5_importing_custom_module.jl

include("person.jl")

# `using .PersonModule` brings exported names into scope
using .PersonModule

p0 = Person("Alice", 30)
greet(p0)

d1 = Dog("Cinna", 2)
greet(d1)

d2 = Dog("Joey", 13)
greet(d2)

c1 = Cat("Cat1", 100)
greet(c1)

File: 1_Review/javascript/5_importing_custom_module.js
// 5) Importing a custom module
// Run: node 5_importing_custom_module.js

import { Person, Dog, Cat } from './person.js';

const p0 = new Person('Alice', 30);
p0.greet();

const d1 = new Dog('Cinna', 2);
d1.greet();

const d2 = new Dog('Joey', 13);
d2.greet();

const c1 = new Cat('Cat1', 100);
c1.greet();
